extension/lib/addProductsToCart.js

```diff
@@ -25,19 +24,21 @@ module.exports = async (context, input) => {
 
   const shopgateCartPipeline = ShopgateCartPipeline.create(context)
   try {
-    await shopgateCartPipeline.addProducts(input.products).catch((err) => {
-      if (err.code === 422) {
-        const message = new ShopgateCartMessage('error', 'This product is not available anymore', 422)
-        return {
-          cartId: '',
-          messages: [message.toJson()]
-        }
-      }
-      return {}
-    })
-  } catch (error) {
-    context.log.error(decorateError(error), 'Failed adding products to cart')
-    throw new Error()
+    await shopgateCartPipeline.addProducts(input.products)
+  } catch (err) {
+    context.log.error(decorateError(err), 'Failed adding products to cart')
+    if (err.code === 422) {
+      let ecartError = new Error('Failed to add product to the cart')
```

Comment:
I expect this to still be a class extending Error as you will be reusing it in other places, e.g. updateProductsInCart call needs to also implement this?
Moreover, why don't you handle this error in the CartExtensionPipeline, they already catch 422 errors there.

Reply:
I handle the exception now in CartExtensionPipeline. The other catch for 422 is completely different and adds an message to the messagesRepository